external-crates/move/crates/move-vm-runtime/src/jit/mod.rs

```diff
@@ -27,3 +27,81 @@ pub fn translate_package(
     };
     execution::translate::package(natives, link_context, opt_package)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::translate_package;
+    use crate::{
+        jit::execution::ast::Package as RuntimePackage, shared::linkage_context::LinkageContext,
+        validation::verification::ast as verif_ast,
+    };
+    use move_binary_format::file_format::empty_module;
+    use move_core_types::{account_address::AccountAddress, language_storage::ModuleId};
+    use move_vm_config::runtime::VMConfig;
+    use std::collections::BTreeMap;
+
+    fn make_verified_empty_package(
+        original_id: AccountAddress,
+        version_id: AccountAddress,
+    ) -> verif_ast::Package {
+        // Minimal valid module
+        let module = empty_module();
+        let module_id: ModuleId = module.self_id();
+
+        // Assemble verification package with a single module and minimal tables
+        verif_ast::Package {
+            original_id,
+            version_id,
+            modules: BTreeMap::from([(module_id, verif_ast::Module { value: module })]),
+            type_origin_table: vec![],
+            linkage_table: BTreeMap::from([(original_id, version_id)]),
+        }
+    }
+
+    fn assert_basic_runtime_pkg(
+        pkg: &RuntimePackage,
+        original_id: AccountAddress,
+        version_id: AccountAddress,
+    ) {
+        assert_eq!(pkg.original_id, original_id);
+        assert_eq!(pkg.version_id, version_id);
+        // One module translated from the single compiled module
+        assert_eq!(pkg.loaded_modules.len(), 1);
+    }
```

Comment:
Are there more things we should check?

I could imagine adding very simple operations that do not get optimized, or even calling simple functions and ensuring the execute the same way.